### PR TITLE
Add REST module

### PR DESCRIPTION
**Pull request by dougwettlaufer**

### CODE REVIEW

**Comment by jorgebay** [comment]
It looks like the integration tests are missing.

**Comment by dougwettlaufer** [comment]
> It looks like the integration tests are missing.

This PR predates the integration tests :). I'll get them added now.

**Review by mpenick — APPROVED** [review]

File: auth-table-based-service/src/main/java/io/stargate/auth/table/AuthTableBasedServiceActivator.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                        log.info("Registering authTableBasedService in AuthTableBasedServiceActivator");
+                        registration = context.registerService(AuthenticationService.class.getName(), authTableBasedService, props);
+                    }
+                    break;
+                case (ServiceEvent.UNREGISTERING):
+                    log.info("Service of type " + objectClass[0] + " unregistered.");
+                    context.ungetService(serviceEvent.getServiceReference());
+                    break;
+                case (ServiceEvent.MODIFIED):
+                    // TODO: [doug] 2020-06-15, Mon, 12:58 do something here...
```
Comment (EricBorczuk) [inline auth-table-based-service/src/main/java/io/stargate/auth/table/AuthTableBasedServiceActivator.java:91]: I saw this comment before in the C2 repo, out of curiosity what _should_ we be doing here? re-getting the service?

**Review by EricBorczuk — APPROVED** [review]
Just had a couple questions, looks good tho

**Pull request merged**